packages/@lwc/ssr-runtime/src/render.ts

```diff
@@ -193,22 +193,26 @@ interface ComponentWithGenerateMarkup extends LightningElementConstructor {
 
 export class RenderContext {
     styleDedupeIsEnabled: boolean;
-    stylesheetToId = new WeakMap<Stylesheet, string>();
     styleDedupePrefix: string;
+    stylesheetToId = new WeakMap<Stylesheet, string>();
     nextId = 0;
 
-    constructor(styleDedupePrefix: string, styleDedupeIsEnabled: boolean) {
-        this.styleDedupeIsEnabled = styleDedupeIsEnabled;
-        this.styleDedupePrefix = styleDedupePrefix;
+    constructor(styleDedupe: string | boolean) {
+        if (styleDedupe || styleDedupe === '') {
+            this.styleDedupePrefix = styleDedupe && '';
+            this.styleDedupeIsEnabled = true;
+        } else {
+            this.styleDedupePrefix = '';
+            this.styleDedupeIsEnabled = false;
+        }
     }
 }
 
 export async function serverSideRenderComponent(
     tagName: string,
     Component: ComponentWithGenerateMarkup,
     props: Properties = {},
-    styleDedupePrefix = '',
-    styleDedupeIsEnabled = false,
+    styleDedupe: string | boolean = '',
```

Comment:
```suggestion
    styleDedupe: string | boolean = false,
```
`""` is falsey so we'd reach the `else` block in the constructor above. But using `""` as the placeholder falsey value seems less clear than using `false`.